packages/docusaurus/src/webpack/utils.ts

```diff
@@ -448,43 +440,8 @@ export function getMinimizer(useSimpleCssMinifier = false): Plugin[] {
         },
       },
     }),
+    new CssMinimizerWebpackPlugin(),
   ];
 
-  if (useSimpleCssMinifier) {
```

Comment:
Why is it removed?

Reply:
1. I couldn't find a proper use case for it
2. The API is different for the other plugin

Reply:
> I couldn't find a proper use case for it

We need to keep this functionality, it is important for better optimization.

Reply:
I'll work on this when I feel less tired.

Reply:
Alright, I wrote up a detailed summary of the issues with this in #4027 @lex111

Reply:
Thanks! But CSS minification is still experimental feature, so could you bring back `useSimpleCssMinifier` condition and related code?

Reply:
Alright.

Reply:
It's strange why, after applying all the minifier options, CSS bundle size still bigger then should be. I mean currently on preview it is 90.5 kB, although it should be 87.1 kB. Any thoughts of this?

Reply:
I have no clue why.  Maybe something just isn't getting post processed.

Reply:
Maybe because cssnano doesn't support PostCSS 8, so appropriate preset doesn't actually work? Can we use PostCSS v7?

Reply:
I guess its possible, I'll give it a try

Reply:
@RDIL it looks like multiple instances of `CssMinimizerPlugin` is not supported. Could you please create issue with bug report in the repository https://github.com/webpack-contrib/css-minimizer-webpack-plugin?

Reply:
I know a way to get around this.

Reply:
As I can see, this workaround apparently didn't work. It might be worth creating issue, I think the authors of this webpack plugin will help us.

Reply:
@alexander-akait any idea why this is happening?

Reply:
hmm, I think I see the problem, why you need two CSS minimizers? Feel free to open an issue

Reply:
@alexander-akait for better minification, we use our `cssnano` preset (based on advanced built-in preset with a couple of other PostCSS plugins) and the `clean-css` library (only its 2 level optimization).

Reply:
anyway it should work, can you open an issue?

Reply:
Done, https://github.com/webpack-contrib/css-minimizer-webpack-plugin/issues/62

Reply:
@lex111 normally I would be a bundle size freak but this 4kb really doesn't seem that awful.  It can be fixed in a future PR instead of blocking this one, and it has actually done a good thing which is that now the Algolia Docsearch copyright header is preserved, which I'm pretty sure is a legal requirement of some sort?  I'm not completely sure.

Reply:
So what I understand is that we can have a single plugin instance with 2 minifiers? (ie using preset + custom cleanCss minifier?)

![image](https://user-images.githubusercontent.com/749374/112323789-64508d00-8cb2-11eb-8b96-809e4b21be33.png)

Reply:
The issue is now closed but `css-minimizer-webpack-plugin` is not released yet. But I think this array syntax existed before, it was just better documented now + add some additional export helpers, so we can probably already fix this

Reply:
Was able to upgrade css minimizer plugin to 2.0, seems the css size remains identical (+5kb).

It's a bit hard to understand why we have this little size increase, it could be anything due to any change in any dependency (including webpack 4->5), and the current css outputs have some significant differences regarding how the CSS but the site keeps working.

I think we should move on and merge this PR asap now, and CSS optimizations PR can be added later, but I'd rather focus on that after we align all deps to postcss8

Reply:
The order of minifiers in the `minify` array does matter, currently it is not correct -- the first element must be `cssnanoMinify`, not cleanCssMinify.

Reply:
ooh good catch you are right!

I actually tried to reverse the order because it was not totally clear by reading the doc.

I'll have to fix this, but I think using the correct order leads to an error 😓 will check that thanks

Note: I think @RDIL may have applied things in the wrong order too when using the minimize callback function. Using the right order could reduce the CSS output finally (if we can make it work)

Reply:
Trying make work minification, but can't figure out why the dependencies in the cssnano preset are not resolving properly, I get an error like this:

```
ReferenceError: _commentRemover is not defined
```

Although I have updated cssnano-preset-advanced to 5.0.0 as well as postcss to v8. What's the matter?

Reply:
Thanks @lex111 , your `require.resolve` trick seems to have worked actually, just needed a few minor changes as I think the preset should now be a function.

The minimizer has some kind of eval that prevents preset imports from working, not sure exactly what's the purpose of that code. Reported here: https://github.com/webpack-contrib/css-minimizer-webpack-plugin/issues/83

It seems the preset is working again and the CSS is now smaller.

Reply:
Btw converted postcss plugins to v8, mostly porting the code from @armano2 from https://github.com/facebook/docusaurus/pull/4404

Reply:
![image](https://user-images.githubusercontent.com/749374/115158239-7e9d4100-a08d-11eb-8057-c2a273a037a5.png)

Reply:
@slorber hooray, you are genius, minification works as before now! :clap: